src/containers/ProjectContainer/TeststepContainer/index.jsx

```diff
@@ -52,81 +49,116 @@ const TeststepContainer = (props) => {
 
     useEffect(() => {
         // get all teststeps, endpoints, headers, payloads
-        dispatch(getTeststepsRequest({project: projectName}));
-        dispatch(getEndpointsRequest({project: projectName}))
-        dispatch(getHeadersRequest({project: projectName}))
-        dispatch(getPayloadsRequest({project: projectName}))
-    }, [projectName])
+        dispatch(getTeststepsRequest({ project: projectName }));
+        dispatch(getEndpointsRequest({ project: projectName }));
+        dispatch(getHeadersRequest({ project: projectName }));
+        dispatch(getPayloadsRequest({ project: projectName }));
+    }, [projectName]);
 
     useEffect(() => {
-        setSelectedItem(teststeps.filter(e => e.id==id)[0]);
-        dispatch(getTestdataRequest({teststep: id}));
-    }, [teststeps, id])
-    
+        setSelectedItem(teststeps.filter(e => e.id == id)[0] || {});
```

Comment:
check if you can use '===' or '==' is used correctly?

Reply:
The type of 'e.id' differ from 'id' because of backend sends the 'id' in form of a string

Reply:
Okay